include/gridtools/stencil_composition/sid/block.hpp

```diff
@@ -0,0 +1,125 @@
+            template <class Stride, class BlockSize>
+            using blocked_stride_type = decltype(block_stride(std::declval<Stride>(), std::declval<BlockSize>()));
+
+            template <class Dim, class BlockedDim>
+            struct generate_strides_f;
+
+            template <class Dim>
+            struct generate_strides_f<Dim, Dim> {
+                using type = generate_strides_f;
+
+                template <class Strides, class BlockMap>
+                auto operator()(Strides const &strides, BlockMap const &) GT_AUTO_RETURN(at_key<Dim>(strides));
+            };
+
+            template <class Dim>
+            struct generate_strides_f<Dim, blocked_dim<Dim>> {
+                using type = generate_strides_f;
+
+                template <class Strides, class BlockMap>
+                auto operator()(Strides const &strides, BlockMap const &block_map) const
+                    GT_AUTO_RETURN(block_stride(at_key<Dim>(strides), at_key<Dim>(block_map)));
+            };
+
+            template <class Sid, class BlockMap>
+            class blocked_sid : public delegate<Sid> {
+                BlockMap m_block_map;
+
+                using strides_map_t = GT_META_CALL(hymap::to_meta_map, GT_META_CALL(strides_type, Sid));
+                using strides_dims_t = GT_META_CALL(meta::transform, (meta::first, strides_map_t));
+
+                template <class MapEntry, class Dim = GT_META_CALL(meta::first, MapEntry)>
+                GT_META_DEFINE_ALIAS(is_strides_dim, meta::st_contains, (strides_dims_t, Dim));
+
+                using block_map_t = GT_META_CALL(
+                    meta::filter, (is_strides_dim, GT_META_CALL(hymap::to_meta_map, BlockMap)));
+                using block_dims_t = GT_META_CALL(meta::transform, (meta::first, block_map_t));
+
+                template <class MapEntry,
+                    class Dim = GT_META_CALL(meta::first, MapEntry),
+                    class BlockSize = GT_META_CALL(meta::second, MapEntry),
+                    class Stride = GT_META_CALL(meta::second, (GT_META_CALL(meta::mp_find, (strides_map_t, Dim))))>
+                GT_META_DEFINE_ALIAS(block, meta::list, (blocked_dim<Dim>, blocked_stride_type<Stride, BlockSize>));
+
+                using blocked_strides_map_t = GT_META_CALL(meta::transform, (block, block_map_t));
+
+                using strides_t = GT_META_CALL(hymap::from_meta_map,
+                    (GT_META_CALL(meta::concat, (strides_map_t, GT_META_CALL(meta::transform, (block, block_map_t))))));
+
+                using original_dims_t = GT_META_CALL(meta::concat, (strides_dims_t, block_dims_t));
+                using blocked_dims_t = GT_META_CALL(get_keys, strides_t);
+
+                template <class L>
+                GT_META_DEFINE_ALIAS(make_generator, meta::rename, (generate_strides_f, L));
+                using generators_t = GT_META_CALL(
+                    meta::transform, (make_generator, GT_META_CALL(meta::zip, (original_dims_t, blocked_dims_t))));
```

Comment:
FYI: no need for `meta::zip` here.  `meta::transform` can accept any number of type lists. the arity of the transformation function should match the number of type lists.